analyzers/src/SonarAnalyzer.CSharp/Rules/UseAwaitableMethod.cs

```diff
@@ -86,7 +87,27 @@ private static WellKnownExtensionMethodContainer BuildWellKnownExtensionMethodCo
         return wellKnownExtensionMethodContainer;
     }
 
-    private static ImmutableArray<ISymbol> FindAwaitableAlternatives(WellKnownExtensionMethodContainer wellKnownExtensionMethodContainer,
+    private static ImmutableArray<Func<IMethodSymbol, bool>> BuildExclusions(Compilation compilation)
+    {
+        var exclusions = ImmutableArray.CreateBuilder<Func<IMethodSymbol, bool>>();
+        if (compilation.GetTypeByMetadataName(KnownType.Microsoft_EntityFrameworkCore_DbSet_TEntity) is not null)
+        {
+            exclusions.Add(x => x.IsAny(KnownType.Microsoft_EntityFrameworkCore_DbSet_TEntity, ExcludedMethodNamesAddAddRange)); // https://github.com/SonarSource/sonar-dotnet/issues/9269
+            exclusions.Add(x => x.IsAny(KnownType.Microsoft_EntityFrameworkCore_DbContext, ExcludedMethodNamesAddAddRange));     // https://github.com/SonarSource/sonar-dotnet/issues/9269
+        }
+        if (compilation.GetTypeByMetadataName(KnownType.FluentValidation_IValidator) is not null)
+        {
+            exclusions.Add(x => x.IsImplementingInterfaceMember(KnownType.FluentValidation_IValidator, "Validate"));   // https://github.com/SonarSource/sonar-dotnet/issues/9339
```

Comment:
`IsImplementingInterfaceMember` is a bit more expensive than a simple `Is` or `IsAny`. I, therefore, change the implementation to follow the same patterns as `BuildWellKnownExtensionMethodContainers`: create the lookup conditionally on compilation start and only add conditions that can potentially be true for the current compilation.